lib/chef/knife/core/subcommand_loader.rb

```diff
@@ -23,7 +23,7 @@ class Knife
     class SubcommandLoader
 
       MATCHES_CHEF_GEM = %r{/chef-[\d]+\.[\d]+\.[\d]+}.freeze
-      MATCHES_THIS_CHEF_GEM = %r{/chef-#{Chef::VERSION}/}.freeze
+      MATCHES_THIS_CHEF_GEM = %r{/chef-#{Chef::VERSION}(-x86-mingw32)?/}.freeze
```

Comment:
I think if you remove some of your invalid tests, and allow it to match to anything like `(-\s*)?(-\s*)?` it should work.
Not sure of the regex is correct, but the idea is optional cpu, optional os